addons/content-index/src/main/java/org/commonjava/indy/content/index/IndexingContentManagerDecorator.java

```diff
@@ -228,25 +228,28 @@ else if ( indexCfg.isAuthoritativeIndex() && store.isAuthoritativeIndex() )
 
             if ( specialPathInfo == null || !specialPathInfo.isMergable() )
             {
-                transfer = getIndexedMemberTransfer( (Group) store, path, TransferOperation.DOWNLOAD,
-                                                     ( member ) -> {
-                                                         try
-                                                         {
-                                                             return delegate.retrieve( member, path );
-                                                         }
-                                                         catch ( IndyWorkflowException e )
-                                                         {
-                                                             logger.error( String.format(
-                                                                     "Failed to retrieve() for member path: %s:%s. Reason: %s",
-                                                                     member.getKey(), path, e.getMessage() ), e );
-                                                         }
-
-                                                         return null;
-                                                     } );
+                for ( StoreKey memberKey : ( (Group) store ).getConstituents() )
+                {
+                    try
+                    {
+                        ArtifactStore member = storeDataManager.getArtifactStore( memberKey );
+                        if ( member != null )
+                        {
+                            transfer = retrieve( member, path, eventMetadata );
+                        }
+                    }
+                    catch ( IndyDataException e )
+                    {
+                        logger.error( String.format( "Failed to lookup store: %s (in membership of: %s). Reason: %s",
+                                                     memberKey, store.getKey(), e.getMessage() ), e );
+                    }
+                }
 
                 if ( transfer != null )
                 {
                     nfc.clearMissing( resource );
+                    logger.debug( "Got transfer from delegate: {} (will index)", transfer );
+                    indexManager.indexTransferIn( transfer, store.getKey() );
```

Comment:
Will this include the originKey information? I'm having a little trouble remembering...